src/controllers/tourController.js

```diff
@@ -36,15 +36,33 @@ exports.fetchSingleTour = (req, res) => {
 }
 
 // book a tour
-exports.store = async (req, res) => {
-	// TODO: get user id from url
-	// instantiating new booking
-	const newBooking = new Booking({...req.body});
-	// saving new booking
-	const bookedTour = await newBooking.save();
-	console.log(bookedTour);
-	// send confirmation email
-	await sendConfirmationEmail(bookedTour, req, res);
+exports.bookTour = async (req, res) => {
+	try {
+		// get user id from url
+		let loggedInUser = await User.findById(req.params.id);
```

Comment:
Logged in user should be from req.user, not req.params. Check line 9 here to further understand how you wrote it in the auth middleware
 https://github.com/reskillamericans/Tour-Info-Group-Node/blob/development/src/middlewares/authenticate.js